distopia/_distopia.pyx

```diff
@@ -99,6 +99,24 @@ def get_n_double_lanes():
     """The number of doubles per register distopia will handle on this system"""
     return GetNDoubleLanes()
 
+# check a variable number of arrays for matching shape and dtype
+def check_arrays(*arrays, results=None):
+    """Check that all arrays are the same length and dtype"""
+    shapes = [a.shape for a in arrays]
+    shape_set = set(shapes)
```

Comment:
```
In [1]: def check_lengths(arrs):
   ...:     lengths = (len(a) for a in arrs)
   ...:     first_length = next(lengths)
   ...:     if not all(l == first_length for l in lengths):
   ...:         raise ValueError
   ...: 
```
A set is kind of ugly here for a single shot, how about a generator to handle the variadic input?

I'm probably fretting over nothing, but rather than being general maybe we should instead be checking these lengths inside a `cdef` regime too?

Reply:
Probably, I think I will open a new PR for this.